src/execution_tree/primitives/primitive_component_base.cpp

```diff
@@ -267,22 +267,33 @@ namespace phylanx { namespace execution_tree { namespace primitives
             return hpx::launch::async;
         }
 
+        // get eval count and execution time of primitive lower and upper threshold
+        // from command line
+        static std::size_t ec_threshold = std::stoul(
+            hpx::get_config_entry("phylanx.eval_count_threshold", "5"));
+        static std::size_t exec_upper_threshold =
+            std::stoul(hpx::get_config_entry(
+                "phylanx.exec_time_upper_threshold", "500000"));
+        static std::size_t exec_lower_threshold =
+            std::stoul(hpx::get_config_entry(
+                "phylanx.exec_time_lower_threshold", "350000"));
+
```

Comment:
Could you extract those into separate functions (similar to the already existing configuration setting)?